### DISCUSSION TITLE
Can Hono run inside a Service Worker in a web browser?

### DISCUSSION THREAD

**Discussion by alexgleason** [honojs/hono#954] [Q&A] [answered]
I'm building a simple Deno compatibility layer to translate API calls from one system into another system. It occurred to me it might be possible to run this code in a ServiceWorker, where the Hono app intercepts fetch requests. Has anyone tried?

**Comment by yusukebe** [comment, marked as answer]
Yes, I've tried it:)

* https://yusukebe.com/posts/2022/service-worker-magic/
* https://github.com/yusukebe/service-worker-magic

**Reply by alexgleason** [reply]
Amazing! Nice work.

**Reply by ajaishankar** [reply]
https://github.com/yusukebe/service-worker-magic?tab=readme-ov-file#short-version

It's mentioned "Short Version: It does not work well."

Shouldn't it be, "it does work well"? 🙂 

Just amazing! Love how simple everything is...

**Reply by yusukebe** [reply]
I forgot why I wrote, "It does not work well," but Hono can work on browsers exactly!
